### PR TITLE
Cleanup pundit policies

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

File: src/api/app/models/package.rb [hunk 1]
```diff
@@ -313,8 +313,13 @@ def master_product_object
...
 
-  def check_write_access(ignoreLock = nil)
-    User.current.can_modify_package? master_product_object, ignoreLock
+  def check_write_access(ignoreLock = nil, user = User.current)
+    user.can_modify_package? master_product_object, ignoreLock
+  end
+
+  # For the sake of having a more speaking method name...
+  def can_be_modified_by?(user)
+    check_write_access(nil, user)
```
Comment (bgeuken) [inline src/api/app/models/package.rb]: We could even replace check_write_access completely. It's only used 3 times in OBS.

File: src/api/app/policies/attrib_policy.rb [hunk 2]
```diff
@@ -39,4 +20,32 @@ def update?
...
+  #   - User
+  #   - Group association of user
+  #   - Role association of user (Only for AttribTypeModifiableBy)
+  def has_permissions_for_modifiables?(attrib_modifiable_bies)
+    attrib_modifiable_bies.to_a.any? do |rule|
+      rule.user == @user ||
+      @user.is_in_group?(rule.group) ||
+        (rule.try(:role) && @user.has_local_role?(rule.role, @record.container))
+    end
+  end
```
Comment (bgeuken) [inline src/api/app/policies/attrib_policy.rb]: This probably could be further refactored by moving it into the modifiable models. But I didn't want to send more effort with out checking if you guys agree with the general approach.

File: src/api/app/policies/attrib_policy.rb [hunk 3]
```diff
@@ -39,4 +20,32 @@ def update?
...
+  private
+
+  # Parses AttribTypeModifiableBy and AttribNamespaceModifiableBy association
+  # collections and checks the users permissions.
+  #
+  # Permission check is based on:
+  #   - User
+  #   - Group association of user
+  #   - Role association of user (Only for AttribTypeModifiableBy)
+  def has_permissions_for_modifiables?(attrib_modifiable_bies)
```
Comment (evanrolfe) [inline src/api/app/policies/attrib_policy.rb]: Maybe I missed this in the pair programming but what is `bies` in `attrib_modifiable_bies` supposed to mean?

Comment (bgeuken) [inline src/api/app/policies/attrib_policy.rb]: Apparently that's how we / rails calls the association. See https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/attrib_type.rb#L17

We already had a laugh in the mobprogramming session 😆

File: src/api/app/models/package.rb (hunk 1 above)
Comment (evanrolfe) [inline src/api/app/models/package.rb]: I say go for it! :+1:

File: src/api/app/policies/attrib_policy.rb (hunk 3 above)
Comment (evanrolfe) [inline src/api/app/policies/attrib_policy.rb]: ahh i see its meant to be the plural of `AttribTypeModifiableBy`... Kind of a tricky situation, better than what I thought before that it was just some made up word lol!

File: src/api/app/policies/attrib_policy.rb (hunk 2 above)
Comment (evanrolfe) [inline src/api/app/policies/attrib_policy.rb]: I think we should keep authorization code in the policy classes? Seems like thats where is belongs?

File: src/api/app/policies/attrib_policy.rb (hunk 3 above)
Comment (bgeuken) [inline src/api/app/policies/attrib_policy.rb]: I'll change it to modifiables like I did for the methodname. Hope that sounds better, it does for me^^

File: src/api/app/policies/attrib_policy.rb (hunk 2 above)
Comment (bgeuken) [inline src/api/app/policies/attrib_policy.rb]: :+1:

And I think I found a nicer way to further simplify:  f4100bd1c4fa004afa09c93cd910ff51edfb32bd

File: src/api/app/models/package.rb [hunk 4]
```diff
@@ -313,13 +313,14 @@ def master_product_object
     product_object || self
   end
 
-  def check_write_access(ignoreLock = nil)
-    User.current.can_modify_package? master_product_object, ignoreLock
+  # For the sake of having a more speaking method name...
```
Comment (evanrolfe) [inline src/api/app/models/package.rb]: Really tiny point but do we need this comment? Once this code is merged I dont think it will really make sense..

Comment (bgeuken) [inline src/api/app/models/package.rb]: Oh, right.

**Comment by evanrolfe** [comment]
Nice work! LGTM

**Pull request merged**